File: Backend/src/main/java/com/example/CheckInApp/dto/request/RegistrationRequest.java
package com.example.CheckInApp.dto.request;

import com.example.CheckInApp.model.FoodPreference;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class RegistrationRequest {

    private Boolean gdprConsent;

    private Boolean photoConsent;

    private FoodPreference foodPreference;

    private Boolean transportNeeded;

    @Size(min = 2, max = 64, message = "Driver name must be between 2 and 64 characters long.")
    private String driverName;

    @Size(min = 5, max = 12, message = "Driver phone number must be between 5 and 12 characters long.")
    private String driverPhoneNumber;

    private Boolean accommodationNeeded;

    @Positive(message = "Number of accommodation days must be positive.")
    private Integer accommodationDays;

}

File: Frontend/src/app/features/events/model/registration-request.ts
import { FoodPreference } from '@features/events/model/food-preference';

export type RegistrationRequest = {
  gdprConsent: boolean;
  photoConsent: boolean;
  foodPreference: FoodPreference | null;
  transportNeeded: boolean | null;
  driverName: string | null;
  driverPhoneNumber: string | null;
  accommodationNeeded: boolean | null;
  accommodationDays: number | null;
};
